Code/Tools/AssetProcessor/native/utilities/ProductOutputUtil.h

```diff
@@ -17,15 +17,22 @@ namespace AssetProcessor
 {
     struct ProductOutputUtil
     {
-        static AZStd::string GetPrefix(AZ::s64 scanfolderId);
+        //! Gets the prefix to apply to products when copying into the cache before being finalized.
+        //! This is different from the final prefix version to avoid conflicts unprefixed assets that may use the same prefix as the final prefix.
+        //! The temp prefix shouldn't show up in any UI so its longer and uglier.
```

Comment:
nit: I understand what you are trying to say with uglier, but it looks odd in a comment. Maybe "Its formatted as "TMP<number>" to make it clear that its an intermediate value.